app/main/views/services.py

```diff
@@ -452,17 +457,29 @@ def update_section_submission(framework_slug, lot_slug, service_id, section_id,
             "services/edit_submission_section.html",
             framework=framework,
             section=section,
+            next_section_name=get_next_section_name(content, section.id),
             service_data=update_data,
             service_id=service_id,
             one_service_limit=has_one_service_limit(lot_slug, framework['lots']),
+            return_to_summary=bool(request.args.get('return_to_summary')),
             errors=errors
         )
 
-    return redirect(url_for(".view_service_submission",
-                            framework_slug=framework['slug'],
-                            lot_slug=draft['lot'],
-                            service_id=service_id,
-                            _anchor=section_id))
+    return_to_summary = bool(request.args.get('return_to_summary'))
+    next_section = content.get_next_editable_section_id(section_id)
+
+    if next_section and not return_to_summary and request.form.get('continue_to_next_section'):
+        return redirect(url_for(".edit_service_submission",
```

Comment:
This is a temporary solution for the third time though – next time DOS submissions open we'll have to switch it back again because we want "back to summary" for DOS. I think if we decide to keep using this flow for G-Cloud we need to support both DOS and G-Cloud ways at once instead of changing this code every iteration.